src/UserService.Business/Commands/Certificate/CreateCertificateCommand.cs

```diff
@@ -2,128 +2,127 @@
-        {
-          imageId = response.Message.Body;
-        }
-      }
-      catch (Exception exc)
-      {
-        _logger.LogError(exc, errorMessage);
-
-        errors.Add(errorMessage);
-      }
-
-      return imageId;
-    }
-
     public CreateCertificateCommand(
       IAccessValidator accessValidator,
       IHttpContextAccessor httpContextAccessor,
       IDbUserCertificateMapper mapper,
-      IUserRepository userRepository,
+      IDbEntityImageMapper entityImageMapper,
+      IImageRepository imageRepository,
       ICertificateRepository certificateRepository,
       IRequestClient<ICreateImagesRequest> rcAddIImage,
-      ICreateImageDataMapper createImageDataMapper,
       ILogger<CreateCertificateCommand> logger)
     {
       _accessValidator = accessValidator;
       _httpContextAccessor = httpContextAccessor;
       _mapper = mapper;
-      _userRepository = userRepository;
+      _entityImageMapper = entityImageMapper;
+      _imageRepository = imageRepository;
       _certificateRepository = certificateRepository;
       _rcImage = rcAddIImage;
       _logger = logger;
-      _createImageDataMapper = createImageDataMapper;
     }
 
-    public OperationResultResponse<Guid> Execute(CreateCertificateRequest request)
+    private async Task<List<Guid>> CreateImages(CreateCertificateRequest request, Guid userId, List<string> errors)
     {
-      if (!(_accessValidator.HasRights(Rights.AddEditRemoveUsers))
-        && _httpContextAccessor.HttpContext.GetUserId() != request.UserId)
+      if (request.Images == null || !request.Images.Any() || request.Images.Contains(null))
       {
-        throw new ForbiddenException("Not enough rights.");
+        return null;
       }
 
-      List<string> errors = new();
+      string logMessage = "Can not add certificate image for user {UserId}. Reason: {Errors}";
 
-      Guid? imageId = GetImageId(request.Image, errors);
+      List<CreateImageData> images = request.Images
+        .Select(i => new CreateImageData(i.Name, i.Content, i.Extension, userId))
+        .ToList();
 
-      if (!imageId.HasValue)
+      try
       {
-        return new OperationResultResponse<Guid>
+        IOperationResult<ICreateImagesResponse> responsedMsg = 
+          (await _rcImage.GetResponse<IOperationResult<ICreateImagesResponse>>(
+            ICreateImagesRequest.CreateObj(images, ImageSource.User))).Message;
+
+        if (responsedMsg.IsSuccess)
         {
-          Status = OperationResultStatusType.Failed,
-          Errors = errors
-        };
+          return responsedMsg.Body.ImagesIds;
+        }
+
+        _logger.LogWarning(logMessage, userId, string.Join(',', responsedMsg.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, userId, exc.Message);
       }
 
-      DbUserCertificate dbUserCertificate = _mapper.Map(request, imageId.Value);
+      errors.Add($"Can not add certificate image for user {userId}. Please try again later.");
```

Comment:
"Can not add images. Please try again later." 
no need to give out information about guid to the user

Reply:
++